Sources/eppo/RuleEvaluator.swift

```diff
@@ -175,95 +240,131 @@ public class FlagEvaluator {
-                        case .lessThan:
-                            return valueDouble < conditionDouble
-                        default:
-                            throw Errors.UnexpectedValue
-                        }
-                    }
-                } catch {
-                    // If stringValue() or doubleValue() throws, or Semver creation fails
+        
+        switch condition.operator {
+        case .greaterThanEqual, .greaterThan, .lessThanEqual, .lessThan:
+            let valueStr = try? value.getStringValue()
+            
+            // If the config is obfuscated, we need to unobfuscate the condition value
+            var conditionValueStr: String? = try? condition.value.getStringValue()
+            if isConfigObfuscated,
+               let cvs = conditionValueStr,
+               let decoded = base64Decode(cvs) {
+                conditionValueStr = decoded
+            }
+            
+            if let valueVersion = valueStr.flatMap(Semver.init), let conditionVersion = conditionValueStr.flatMap(Semver.init) {
+                // If both strings are valid Semver strings, perform a Semver comparison
+                switch condition.operator {
+                case .greaterThanEqual:
+                    return valueVersion >= conditionVersion
+                case .greaterThan:
+                    return valueVersion > conditionVersion
+                case .lessThanEqual:
+                    return valueVersion <= conditionVersion
+                case .lessThan:
+                    return valueVersion < conditionVersion
+                default:
+                    return false
+                }
+            } else {
+                // If either string is not a valid Semver, fall back to double comparison
+                guard let valueDouble = try? value.getDoubleValue() else {
+                    return false
+                }
+                
+                // If the config is obfuscated, we need to unobfuscate the condition value
+                var conditionDouble: Double
+                if isConfigObfuscated,
+                   let cvs = conditionValueStr,
+                   let doubleValue = Double(cvs) {
+                    conditionDouble = doubleValue
+                } else if let doubleValue = try? condition.value.getDoubleValue() {
+                    conditionDouble = doubleValue
+                } else {
+                    return false
+                }
+                
+                switch condition.operator {
+                case .greaterThanEqual:
+                    return valueDouble >= conditionDouble
+                case .greaterThan:
+                    return valueDouble > conditionDouble
+                case .lessThanEqual:
+                    return valueDouble <= conditionDouble
+                case .lessThan:
+                    return valueDouble < conditionDouble
+                default:
```

Comment:
Repeated code. Not a huge deal, but since the semvers seem to have overloaded comparison operators, I wonder if we can build the two operands--either semver or numeric--and then have this logic just once.